oracle/blockeligibilityvalidator.go

```diff
@@ -37,45 +36,37 @@ func NewBlockEligibilityValidator(committeeSize, genesisActiveSetSize uint32, la
 }
 
 func (v BlockEligibilityValidator) BlockSignedAndEligible(block *types.Block) (bool, error) {
+	var activeSetSize uint32
+	var vrfPubkey []byte
+	var genesisNoAtx bool
 
-	//todo remove this hack when genesis is handled
-	if block.Layer().GetEpoch(v.layersPerEpoch) == 0 {
+	epochNumber := block.LayerIndex.GetEpoch(v.layersPerEpoch)
+	if epochNumber == 0 {
+		v.log.With().Warning("skipping epoch 0 block validation.",
+			log.BlockId(block.Id().String()), log.LayerId(block.LayerIndex.Uint64()))
 		return true, nil
 	}
-
-	//check block signature and is identity active
-	pubKey, err := ed25519.ExtractPublicKey(block.Bytes(), block.Sig())
-	if err != nil {
-		return false, err
+	if block.ATXID == *types.EmptyAtxId {
+		if !epochNumber.IsGenesis() {
+			return false, fmt.Errorf("no associated ATX in epoch %v", epochNumber)
+		}
+		genesisNoAtx = true
+	} else {
```

Comment:
looks like you can escape right after "genesisNoAtx = true"
is there a reason for doing it later ?

Reply:
so @barakshani explained why its like this, i think the flows in this method are a little complex 
when we handle the genesis flow we should try and refactor this